msal/src/main/java/com/microsoft/identity/nativeauth/statemachine/results/MFAResult.kt

```diff
@@ -0,0 +1,45 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+package com.microsoft.identity.nativeauth.statemachine.results
+
+import com.microsoft.identity.nativeauth.AuthMethod
+import com.microsoft.identity.nativeauth.statemachine.states.MFARequiredState
+
+interface MFARequiredResult: Result {
+    class VerificationRequired(
+        override val nextState: MFARequiredState,
+        val codeLength: Int,
+        val sentTo: String,
+        val channel: String,
+    ) : MFARequiredResult, Result.SuccessResult(nextState = nextState)
+
+    class SelectionRequired(
+        override val nextState: MFARequiredState,
+        val authMethods: List<AuthMethod>
+    ) : MFARequiredResult, MFAGetAuthMethodsResult, Result.SuccessResult(nextState = nextState)
+}
+
+interface MFASubmitChallengeResult : Result
```

Comment:
Where can I see the subtype of this result?

Reply:
In IDE, click on "implements" arrow. In the PR, look for classes/interfaces that have `: MFASubmitChallengeResult`.

<img width="884" alt="image" src="https://github.com/user-attachments/assets/37cada19-bf8c-45bc-bee8-563a5dbcb7e2">